### ISSUE TITLE
MaterialModal.addCloseHandler parameter is not used

### ISSUE THREAD

**Issue by jzaruba** [GwtMaterialDesign/gwt-material#374]
This is what I see in the source code:
```
    @Override
    public HandlerRegistration addCloseHandler(final CloseHandler<MaterialModal> handler) {
        return this.addHandler(new CloseHandler<MaterialModal>() {
            @Override
            public void onClose(CloseEvent<MaterialModal> event) {

            }
        }, CloseEvent.getType());
    }
```
Instead of registering the handler the method creates a new, empty one.
Naturally the modal does not report it being closed.

**Comment by gilberto-torrezan** [comment]
This is fixed on branch 1.5.2: https://github.com/GwtMaterialDesign/gwt-material/blob/release_1.5.2/gwt-material/src/main/java/gwt/material/design/client/ui/MaterialModal.java

**Comment by jzaruba** [comment]
Damn! I have **just** forked, fixed and pushed it (to my fork)... Could have been my first pull request here and 2nd ever! :P
Anyways, awesomeness, thanks :)
